sklearn/tests/test_metadata_routing.py

```diff
@@ -844,3 +847,77 @@ def test_metadata_routing_get_param_names():
     assert router._get_param_names(
         method="fit", return_alias=True, ignore_self=True
     ) == router._get_param_names(method="fit", return_alias=False, ignore_self=True)
+
+
+def test_method_generation():
+    # Test if all required request methods are generated.
+
+    class SimpleEstimator(BaseEstimator):
```

Comment:
I am wondering if we could move this estimator in `sklearn.utils._testing` next to the `Minimal*` estimator and test it.

Reply:
I think after this PR we can slowly figure out what the *common* test estimators are and move them there. They have evolved quite a bit over time.